src/detect-engine-build.c

```diff
@@ -1139,6 +1139,14 @@ typedef enum {
 
 static int CreateGroupedPortListCmpCnt(const DetectPort *a, const DetectPort *b)
 {
+    if (a->sh->init->score < b->sh->init->score) {
+        return 1;
+    } else if (a->sh->init->score > b->sh->init->score) {
+        return -1;
+    }
+
+    /* if scores are equal, check sig cnt */
+
```

Comment:
This is weird. The list at this point should only belong to one particular score as a part of the optimizations I did earlier. 🤔  Checking..

Reply:
There was a typo in what I had done. 🤦🏽‍♀️ 
Fixing that is enough and we don't need to check the scores again. :) See https://github.com/OISF/suricata/pull/10500